enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,365 @@
+  their own plugins.
+* Provide a single command pre-configured for full report gathering.
+  In order to have a single and simple command to gather the report, provide a
+  profile configuration file for `sos` to run all relevant plugins in one shot.
+* Must not leak sensitive data: IPs, hostnames, passwords, certificates, etc.
+  The /var/lib/microshift directory holds certificates and etcd data, therefore
+  it must be skipped.
+
+Existing debugging tools in use by CEE include [omg](https://github.com/kxr/o-must-gather)
+and [omc](https://github.com/gmeghnag/omc). These provide a CLI like experience
+in terms of getting information from a live cluster, ressembling `oc` commands,
+but coming from a must-gather report. Since they were born out of must-gather,
+they expect the same file structure layout. The must-gather file system
+hierarchy is driven by `oc adm inspect` commands, gathering cluster resources
+data that these tools read upon later.
+
+To achieve the same results the most simple approach is to include the `oc`
+binary with MicroShift as a dependency. The plugin can then call
+`oc adm inspect` to produce the output required. There is no API between the
+plugin and MicroShift, easing maintainability and allowing enough flexibility
+for the future.
+
+### Workflow Description
+> _Disclaimer:_ Workflows described here illustrate manual processes.
+
+Assuming from now on MicroShift has already been installed in a R4E/RHEL based
+system and there has been some kind of issue.
+
+1. MicroShift administrator notices an issue/wants to get a system report.
+1. The administrator will log in to the failing host.
+1. The administrator will perform the debug report creation procedure:
+```bash
+sudo sos report --config-file /etc/microshift/report.conf
+```
+
+For CI jobs the same procedure applies. This shall be automated upon failure.
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are four main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. CEE having a different tool.
+1. Exposing sensitive data.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore backwards compatibility with the plugin must be ensured.
+
+MicroShift will include `sos` as a dependency, tying them together in case of
+upgrades/downgrades.
+
+The only moment where there could be a mismatch is right after upgrading
+`sos`, and failing to upgrade `microshift`. In this case the gathering tool
+will be at a higher version than `microshift`. The opposite situation also
+holds true. Extra care must be taken in order to support these situations by
+using the minimum amount of hardcoded names and assumptions on the system. If
+there is a generic way of gathering information, it should be used.
```

Comment:
must-gather took a policy that it will try to gather as much as possible data, without failing. I'd suggest do prevent the problems with compatibility to use the same approach, report problem, but continue gathering data.